File: Python/Queue/Circular_queue.py
def enqueue(x):
    global rear
    if isFull()==False:
        rear = (rear + 1) % MAX_QUEUE_SIZE
        queue.insert(rear, x)

def dequeue():
    global front, rear
    if isEmpty() == False:
        front = (front + 1) % MAX_QUEUE_SIZE
        returnValue = queue[front]
        queue[front] = " "
        return returnValue
    else:
        print("queue is Empty")

def peek():
    return queue[0]

def isSize():
    return len(queue)

def isEmpty():
    if rear == front:
        return True
    else:
        return False

def isFull():
    if front == (rear + 1) % MAX_QUEUE_SIZE:
        print("queue is full")
        return True
    else:
        return False

def init():
    global front, rear

    front = -1
    rear = -1
    queue.clear()

##-------------------- test Code  --------------------

MAX_QUEUE_SIZE = 7
front = -1
rear = -1
queue = []

init()
enqueue(1)
print(queue)
enqueue(3)
print(queue)
dequeue()
print(queue)
enqueue(5)
print(queue)
enqueue(8)
print(queue)

enqueue(7)
print(queue)
enqueue(9)
print(queue)
enqueue(11)
print(queue)
enqueue(4)
print(queue)
dequeue()
print(queue)

File: Java/src/Queue/Circular_queue.java
package Queue;
import java.util.Arrays;

public class Circular_queue {

    private int[] array;
    private int capacity;
    private int rear = 0, front = 0;
    private char flag = 0;

    public ArrayCircularQueue_2(int capacity) {
        this.capacity = capacity;
        this.array = new int[capacity];
    }

    public ArrayCircularQueue_2() {
        this.capacity = 5;
        this.array = new int[capacity];
    }

    public int size() {
        return (capacity * flag + (rear - front));
    }

    public boolean isFull() {
        return size() == capacity;
    }

    public boolean isEmpty() {
        return size() == 0;
    }

    public void enqueue(int data) {
        if (isFull())
            throw new ArrayIndexOutOfBoundsException();

        this.array[rear] = data;
        this.rear = (this.rear + 1) % capacity;

        if (front >= rear) flag = 1;
        else flag = 0;
    }

    public int dequeue() {
        if (isEmpty())
            throw new ArrayIndexOutOfBoundsException();

        int data = this.array[front];
        this.array[front] = 0;
        this.front = (this.front + 1) % capacity;

        if (front > rear) flag = 1;
        else flag = 0;
        return data;
    }

    public int peek() {
        return this.array[front];
    }

    @Override public String toString() {
        return Arrays.toString(array) + " ,  size : " + size();
    }
}